worker/uniter/runner/context/contextfactory.go

```diff
@@ -214,6 +214,16 @@ func (f *contextFactory) HookContext(hookInfo hook.Info) (*HookContext, error) {
 			relation.cache.InvalidateMember(hookInfo.RemoteUnit)
 		}
 		hookName = fmt.Sprintf("%s-%s", relation.Name(), hookInfo.Kind)
+		// Update the private address on the context from the relation settings
```

Comment:
I think there are charms that override private-address. I'm guessing the idea is to not redo the logic we've already done to figure out what to use for privateAddress. But I'd rather we did something like:
PrivateAddressForHook rather than reading the relation settings to get the same content.